packages/next/client/use-intersection.tsx

```diff
@@ -110,18 +120,16 @@ function createObserver(options: UseIntersectionObserverInit): Observer {
     root: options.root || null,
     margin: options.rootMargin || '',
   }
-  let existing = idList.find(
+  const existing = idList.find(
     (obj) => obj.root === id.root && obj.margin === id.margin
   )
-  let instance
+  let instance: Observer | undefined
+
   if (existing) {
     instance = observers.get(existing)
-  } else {
-    instance = observers.get(id)
-    idList.push(id)
-  }
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#key_equality

> Key equality (of `Map`) is based on the [sameValueZero](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#same-value-zero_equality) algorithm.

`id` is a newly created literal object, so `observers.get(id)` will always return `undefined`.

Reply:
I see now. Oddly enough, this code was added in #33933 to fix an issue with lazy loading so I'm surprised it ever worked. cc @11koukou since this makes me think the test added in #33933 is testing the wrong thing.